### DISCUSSION TITLE
Why there is no composer.lock files in the Symfony repos?

### DISCUSSION THREAD

**Discussion by madigi** [symfony/symfony#53813] [Q&A] [answered]
Hi people

Just wondering... why there is no composer.lock file at all in the symfony/* repos? I've check some branchs and tags but didn´t see that files

I've always thought  it's a good practice but maybe I'm wrong about this

Any thoughts?

Regards.

**Comment by stof** [comment, marked as answer]
It is a good practice for *projects* to ensure reproducible installs.
But for libraries, this is not the case at all. 

In a library, the lock file would be used only for contributors installing dependencies when that library is the root package. And for most libraries, we actually don't want to have locked dependencies there. Instead, we want to install the latest (or sometimes lowest) version of the dependencies that are compatible with the current platform to test what our users would be using (and simplify a lot the management of running tests on a wide range of PHP versions).
Thus, in the case of Symfony specifically, contributors would not even work in the repositories of each component but in the mono-repos.

The few places where the public Symfony repos contain a _project_ do have a committed lock file. See for instance the source code for the [ux.symfony.com website](https://github.com/symfony/ux/tree/2.x/ux.symfony.com) or the [demo project](https://github.com/symfony/demo)

**Comment by madigi** [comment]
Thanks @stof for taking your time to write down this elaborated response :)

I agree in the projects points of view to ensure reproducible installs. I'd say this is the most extended approach nowadays

I see you are a member of Symfony, Doctrine and other great organizations so then maybe you can also explain to me: 

As opposite of Symfony ecosystem, doctrine/* ecosystem do has composer.lock files in their tag releases. Why?

Regards.

**Reply by stof** [reply]
In which tag releases ? Neither `doctrine/orm` nor `doctrine/dbal` nor `doctrine/doctrine-bundle` have a committed lock file. I haven't checked other repositories.

**Comment by madigi** [comment]
You are right @stof. Yesterday I check the GitHub tags and assumed the composer.lock files were included to the release but they are not

Thank you so much for your time and clarify to me all this mess

Regards.
